src/core/mac/mac.cpp

```diff
@@ -340,6 +340,10 @@ void Mac::PerformEnergyScan(void)
     }
     else
     {
+        if(!GetRxOnWhenIdle())
+        {
+            SuccessOrAssert(Get().Receive(mScanChannel));
```

Comment:
Should we do:
```suggestion
            mLinks.Receive(mScanChannel);
```

Reply:
Hi @abtink,

Yes, We can do this way as well.